FccCMakeGuide.md

```diff
@@ -58,11 +56,31 @@ Note that when changing the name of a property of an algorithm or a
 tool, `  make` (and not only `  make packagename` ) needs to be run for
 Gaudi to be aware of the change.
 
-### []() Using an internal library
+### CTest in FCCSW
+
+FCCSW also uses the cmake for integration tests.
+This is described in detail in the documentation page on [adding tests to FCCSW](https://github.com/HEP-FCC/FCCSW/blob/master/doc/AddingTestsToFCCSW.md).
+
+### Using an internal library
+
+Once Gaudi is notified that a certain subdirectory is needed by invoking `gaudi_depends_on_subdir`, internal libraries defined in this subdirectory can be used by simply adding them to the list of `INCLUDE_DIRS` and `LINK_LIBRARIES`. An example would be the way the internal library `DetCommon` is used by the module [`Test/TestGeometryPlugins`](https://github.com/HEP-FCC/FCCSW/blob/master/Test/TestGeometry/CMakeLists.txt) in FCCSW.
+
+
+
+### Building a new Gaudi module
+
+The best way is to look at existing modules in FCCSW for inspiration. The syntax to declare the module [`TestGeometryPlugins`](https://github.com/HEP-FCC/FCCSW/blob/master/Test/TestGeometry/CMakeLists.txt), for example, is:
+
+```
+gaudi_add_module(TestGeometryPlugins
+                 src/components/*.cpp
+                 INCLUDE_DIRS Geant4 FWCore SimG4Interface SimG4Common DetInterface DetCommon TestGeometry
+                 LINK_LIBRARIES GaudiKernel FWCore Geant4 DetCommon TestGeometry)
+
+```
+
+### Using an external library
 
-### []() Building a new Gaudi module
+This can be done using the standard cmake command [find_package](https://cmake.org/cmake/help/v3.0/command/find_package.html). See [Colins CMake examples](https://github.com/cbernet/cmake-examples) for details.
 
```

Comment:
Can you add something on the slightly special cases within FCCSW? I'm thinking of ROOT and DD4hep when you have dependencies on certain sub-packages.

Btw, is there a good link that lists which sub-packages from ROOT you need for which classes / functionality?